provider/ec2/local_test.go

```diff
@@ -1358,6 +1363,42 @@ func (t *localServerSuite) testStartInstanceAvailZoneOneConstrained(c *gc.C, run
 	c.Assert(azArgs, gc.DeepEquals, []string{"test-available", "test-available2"})
 }
 
+func (t *localServerSuite) TestStartInstanceWithImageIDErr(c *gc.C) {
+	env := t.prepareAndBootstrap(c)
+
+	expectedImageID := aws.String("ami-1234567890")
+	params := environs.StartInstanceParams{
+		ControllerUUID: t.ControllerUUID,
+		Constraints:    constraints.Value{ImageID: expectedImageID},
+	}
+
+	t.PatchValue(ec2.RunInstances, func(e ec2.Client, ctx context.ProviderCallContext, ri *awsec2.RunInstancesInput, callback environs.StatusCallbackFunc) (resp *awsec2.RunInstancesOutput, err error) {
+		return nil, amiNoImageErr
```

Comment:
This is a bit weird. 

This err hard codes an ami id, which is different from the one which is passed in.

It would probably be easier to maintain/read if instead of hard coding the value here, you constructed it using the ImageId in ri

Something like
```
return nil, &smithy.GenericAPIError{
  Code: "InvalisAMIID",
  Message: fmt.Sprintf("The image id '[%s]' does not exist", ri.ImageId)
}

Reply:
sure! you are right, I added it to the test.